wasmcloud_host/README.md

```diff
@@ -1,19 +1,172 @@
-  * Start Phoenix endpoint with `mix phx.server`
+1. Install the latest versions of Rust and Go
 
-Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
+1. Since the build of `wasmcloud-otp` will fail on version `16.xx.yy` of NodeJs, you should either install, or switch to, version `14.xx.yy` of NodeJS.
 
-Ready to run in production? Please [check our deployment guides](https://hexdocs.pm/phoenix/deployment.html).
+    Typically, a NodeJS version management tool such as `nvm` or `fnm` is used:
 
-## Learn more
+    E.G. To install Fast Node Manager (`fnm`):
 
-  * Official website: https://www.phoenixframework.org/
-  * Guides: https://hexdocs.pm/phoenix/overview.html
-  * Docs: https://hexdocs.pm/phoenix
-  * Forum: https://elixirforum.com/c/phoenix-forum
-  * Source: https://github.com/phoenixframework/phoenix
+    ```console
+    $ cargo install fvm
+    ```
+
+    Install the latest version of NodeJS version 14
+
+    ```console
+    $ fnm install 14
+    Installing Node v14.17.6
+    ```
+
+    Make NodeJS version 14 the default
+
+    ```console
+    $ fnm use 14
+    Using Node v14.17.6
+    ```
+
+1. Clone the repo `wasmCloud/wasmcloud-otp` into some suitable development directory
+
+    ```console
+    $ git clone https://github.com/wasmCloud/wasmcloud-otp.git
+    ```
+
+1. Install and start `nats-server` using the JetStream option
+
+    ```console
+    $ brew install nats-server
+    $ nats-server -js
+    ```
+
+This terminal window is now occupied with the `nats-server` console logs
+
+## Local Server Build
+
+In a new terminal, change into the `wasmcloud-otp` folder, then change into the `assets` folder underneath `wasmcloud_host`
+
+1. `cd wasmCloud/wasmcloud-otp/wasmcloud_host`
+
+1. Run `npm install`.  As long as you have switched to version `14.xx.yy` of NodeJs, this should run successfully
+
+1. Change up one directory level and run `make run-interactive`.  This will build the wasmcloud server and connect it to your already-started NATS server
+
+1. This terminal window now shows both an Elixir command prompt and from time to time, any wasmcloud console logs
+
+1. You can now point your browser to (localhost:4000)[localhost:4000] to see the wasmcloud dashboard
+
+## Building Capability Providers
+
+In the `capability_providers` folder, there are two simple capability providers: `httpserver` and `keyvalue`
+
+Both providers must be built using the supplied `Makefile`, so first change into the `capability_providers` directory
+
+### Building `httpserver`
+
+The `httpserver` provider has been written in Go and can be built using the command:
+
+```console
+$ make httpserver-<os>
+```
+
+Where `<os>` is either `linux`, `mac` or `windows`
+
+E.G. To build the Mac `httpserver` provider
+
+```console
+$ make httpserver-mac
+mkdir -p ./_build
+cd httpserver/go && \
+        GOOS=darwin go build -o httpserver-mac && \
+        mv httpserver-mac ../../_build/httpserver-mac
+```
```

Comment:
does this actually give you something that you can start in the washboard gui, or do you need to `make par-httpserver` and upload the whole bundle?